panic: close of closed channel

When i run 

```golang
	enc := gojay.Stream.NewEncoder(buf).NConsumer(10).CommaDelimited()
	go enc.EncodeStream(processListCh)
	<- enc.Done()
```

in a loop of 1 sec interval, i manage to get 

```
panic: close of closed channel

goroutine 591 [running]:
.../vendor/github.com/francoispqt/gojay.(*StreamEncoder).Cancel(0xc00014c570, 0x0, 0x0)
        .../vendor/github.com/francoispqt/gojay/encode_stream.go:133 +0xa3
...../vendor/github.com/francoispqt/gojay.consume(0xc00014c570, 0xc00014c720, 0x140f2a0, 0xc000122300)
        ..../vendor/github.com/francoispqt/gojay/encode_stream.go:199 +0xeb
created by .../vendor/github.com/francoispqt/gojay.(*StreamEncoder).EncodeStream
      ..../vendor/github.com/francoispqt/gojay/encode_stream.go:53 +0x1b3
exit status 2
```

This is caused by 
https://github.com/francoispqt/gojay/blob/90d953358b68936b4791b8db205b3a5918b1b7a5/encode_stream.go#L124-L136

on line 133, however i've not managed to locate how it got closed right after the `<-done` check and before the `close()` call. However, i can constantly reproduce this 🤔 I have a feeling there should be a  mutex with write lock to prevent this from happening.

--- a/encode_stream.go
+++ b/encode_stream.go
@@ -125,14 +125,15 @@ func (s *StreamEncoder) Value(key interface{}) interface{} {
 //
 // After calling cancel, Done() will return a closed channel.
 func (s *StreamEncoder) Cancel(err error) {
-	s.mux.RLock()
+	s.mux.Lock()
+	defer s.mux.Unlock()
+	
 	select {
 	case <-s.done:
 	default:
 		s.err = err
 		close(s.done)
 	}
-	s.mux.RUnlock()
 }
 
 // AddObject adds an object to be encoded.

